tools/resourcedocsgen/overlays/kubernetes/overlays.json

```diff
@@ -0,0 +1,291 @@
+                    "description": "Location of your Helm config. Overrides $HELM_HOME (default \"/Users/abc/.helm\")."
+                },
+                "keyFile": {
+                    "type": "string",
+                    "description": "Identify HTTPS client using this SSL key file."
+                },
+                "keyring": {
+                    "type": "string",
+                    "description": "Keyring containing public keys (default “/Users/abc/.gnupg/pubring.gpg”)."
+                },
+                "password": {
+                    "type": "string",
+                    "description": "Chart repository password."
+                },
+                "prov": {
+                    "type": "string",
+                    "description": "Fetch the provenance file, but don’t perform verification."
+                },
+                "repo": {
+                    "type": "string",
+                    "description": "Chart repository url where to locate the requested chart."
+                },
+                "untar": {
+                    "type": "boolean",
+                    "description": "If set to false, will leave the chart as a tarball after downloading."
+                },
+                "untardir": {
+                    "type": "string",
+                    "description": "If untar is specified, this flag specifies the name of the directory into which the chart is expanded (default \".\")."
+                },
+                "username": {
+                    "type": "string",
+                    "description": "Chart repository username."
+                },
+                "verify": {
+                    "type": "string",
+                    "description": "Verify the package against its signature."
+                },
+                "version": {
+                    "type": "string",
+                    "description": "Specific version of a chart. Without this, the latest version is fetched."
+                }
+            }
+        }
+    },
+    "resources": {
+        "kubernetes:apiextensions.k8s.io:CustomResource": {
+            "description": "CustomResource represents an instance of a CustomResourceDefinition (CRD). For example, the\n CoreOS Prometheus operator exposes a CRD `monitoring.coreos.com/ServiceMonitor`; to\n instantiate this as a Pulumi resource, one could call `new CustomResource`, passing the\n `ServiceMonitor` resource definition as an argument.",
+            "type": "object",
+            "properties": {
+                "apiVersion": {
+                    "type": "string",
+                    "description": "APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources",
+                    "const": "admissionregistration.k8s.io/v1"
```

Comment:
Yes, this is a copy/paste error. I'll fix it. But should I actually remove the `const` property here? Are users actually allowed to pass `apiVersion` and `kind` for this resource since it is a _Custom_ resource and therefore, won't have a pre-defined value unlike it is for the other k8s resources?

Reply:
Yes, this field is set by the user.

Reply:
Ah ok. I guess I need to update the generator to remove the hard-coded exclusions for `apiVersion` and `kind` and switch to honoring the `constValue` property.

Reply:
Alright I have updated the PR https://github.com/pulumi/pulumi/pull/4426 to honor `ConstValue` rather than specifically excluding `apiVersion` and `kind`.